Source/WebKit/Shared/Cocoa/MobileGestaltParameters.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "MobileGestaltParameters.h"
+
+#if PLATFORM(IOS_FAMILY)
+
+#include "Logging.h"
+#include <pal/spi/ios/MobileGestaltSPI.h>
+
+namespace WebKit {
+
+void MobileGestaltParameters::initialize()
+{
+    mainScreenScale = MGGetFloat32Answer(kMGQMainScreenScale, 0);
+    mainScreenPitch = MGGetSInt32Answer(kMGQMainScreenPitch, 0);
+    mainScreenClass = MGGetSInt32Answer(kMGQMainScreenClass, MGScreenClassPad2);
+    appleInternalInstallCapability = MGGetBoolAnswer(kMGQAppleInternalInstallCapability);
+    iPadCapability = MGGetBoolAnswer(kMGQiPadCapability);
+    deviceName = adoptCF(static_cast<CFStringRef>(MGCopyAnswer(kMGQDeviceName, nullptr))).get();
+    deviceClassNumber = MGGetSInt32Answer(kMGQDeviceClassNumber, MGDeviceClassInvalid);
+    hasExtendedColorDisplay = MGGetBoolAnswer(kMGQHasExtendedColorDisplay);
+    deviceCornerRadius = MGGetFloat32Answer(kMGQDeviceCornerRadius, 0);
+    supportsForceTouch = MGGetBoolAnswer(kMGQSupportsForceTouch);
+    bluetoothCapability = MGGetBoolAnswer(kMGQBluetoothCapability);
```

Comment:
It would be good if we could get Bluetooth knowledge out of the WCP. Seems unlikely there are any cases where the WebContent process should seek this information. Seems like the UIProcess and GPUProcess would be the right places to need this.

But perhaps MobileGestalt fails in some way if this isn't populated accurately.